src/nft/utils/pooledAssets.ts

```diff
@@ -32,6 +33,50 @@ const calculateScaledPrice = (currentPrice: BigNumber, poolFee: BigNumber): BigN
   return currentPrice.add(protocolFee).add(tradeFee)
 }
 
+const calcSudoSwapLinearBondingCurve = (currentPrice: BigNumber, delta: BigNumber, position = 0): BigNumber => {
+  for (let i = 0; i <= position; i++) {
+    currentPrice = currentPrice.add(delta)
+  }
+
+  return currentPrice
+}
+
+const calcSudoSwapExponentialBondingCurve = (currentPrice: BigNumber, delta: BigNumber, position = 0): BigNumber => {
+  for (let i = 0; i <= position; i++) {
+    currentPrice = currentPrice.mul(delta).div(BigNumber.from(PRECISION))
+  }
+
+  return currentPrice
+}
+
+const calcSudoSwapXykBondingCurve = (
+  currentPrice: BigNumber,
+  sudoSwapPool: Pool,
+  position = 0
+): BigNumber | undefined => {
+  let virtualTokenBalance = BigNumber.from(sudoSwapPool.spotPrice)
+  let virtualNFTBalance = BigNumber.from(sudoSwapPool.delta)
+
+  if (virtualNFTBalance.sub(BigNumber.from(1)).gt(BigNumber.from(0))) {
+    currentPrice = virtualTokenBalance.div(virtualNFTBalance.sub(BigNumber.from(1)))
+  } else {
+    return undefined
```

Comment:
I think this undefined might be causing a crash when trying to add past the available amount. Should we return 0 instead?
![xyk](https://user-images.githubusercontent.com/11512321/211913213-f02988dc-bdf0-4973-bdd2-dff5df000c25.gif)

Reply:
its actually returning `0` after the refactor